apps/src/block_utils.js

```diff
@@ -384,13 +384,15 @@ const DUMMY_INPUT = 'dummy';
  *   options.
  * @param {BlockValueType} args[].type For value inputs, the type required. Use
  *   BlockValueType.NONE to accept any block.
+ * @params {string[]} strictTypes Input/output types that are always configerd
+ *   with strict type checking.
  *
  * @returns {Object[]} a list of labeled inputs. Each one has the same fields
  *   as 'args', but additionally includes:
  * @returns {string} return[].mode Either 'dropdown', 'value', or 'dummy'
  * @returns {string} return[].label Text to display to the left of the input
  */
-const determineInputs = function (text, args) {
+const determineInputs = function (text, args, strictTypes=[]) {
```

Comment:
Are the updates to this file a sign that "custom" block parsing should move to the blockly repo at some point?

Reply:
Seems reasonable.